pytensor/xtensor/shape.py

```diff
@@ -69,3 +69,75 @@ def stack(x, dim: dict[str, Sequence[str]] | None = None, **dims: Sequence[str])
             )
         y = Stack(new_dim_name, tuple(stacked_dims))(y)
     return y
+
+
+class UnStack(XOp):
+    __props__ = ("old_dim_name", "unstacked_dims", "unstacked_lengths")
```

Comment:
Seems like nothing requires "unstacked_lengths", to be constant/symbolic. So we could parametrize this Op just with `("old_dim_name", "unstacked_dims")` and pass "unstacked_lengths" to `make_node`. We can convert those to scalar TensorVariables `as_tensor(x, ndim=0)` and check that the `dtype` is integer.

Everything in the rewrite with reshape would work the same, but we would extract them from `node.inputs[1:]`

Reply:
CC @OriolAbril

Reply:
Thanks for the pointers, I'll try to make the updates

Reply:
I have tried this (I believe) and ran into some conversion issues. I will try tomorrow a bit more